geopandas/_vectorized.py

```diff
@@ -1095,10 +1095,21 @@ def bounds(data):
 #
 
 
-def transform(data, func):
+def transform(data, func, include_z=None):
     if compat.USE_SHAPELY_20:
-        coords = shapely.get_coordinates(data)
-        new_coords = func(coords[:, 0], coords[:, 1])
+        if include_z is None:
+            # if not specified explicitly, try to infer from first geometry
+            include_z = False
```

Comment:
You can have mixed arrays. If you had your first geometry Point Z and the second Point, the second would result in Point Empty after the transformation.